### PR TITLE
ELPP-3164 Logging setup

### PR DESCRIPTION
**Pull request by giorgiosironi**
Depends on https://github.com/elifesciences/profiles-formula/pull/10 to make build pass.

- Logging all Flask-handled errors
- JsonFormatter with custom fields
- Added `/error` for error reproducibility
- Allow configuration of level, path

To be done separately: logging of `requests` (standard options are ugly).

```
{
 "exc_info": {
   "message": "Exception: Intentional error",
   "class": "RuntimeError",
   "trace": [
     "  File
\"/srv/profiles/venv/lib/python3.5/site-packages/flask/app.py\", line 1612,
in full_dispatch_request\n    rv = self.dispatch_request()\n",
     "  File
\"/srv/profiles/venv/lib/python3.5/site-packages/flask/app.py\", line 1598,
in dispatch_request\n    return
self.view_functions[rule.endpoint](**req.view_args)\n",
     "  File \"./profiles/api/ping.py\", line 18, in _error\n    raise
RuntimeError('Intentional error')\n"
   ]
 },
 "levelname": "ERROR",
 "asctime": "2017-09-19 16:02:28,090",
 "message": "Intentional error"
}

### CODE REVIEW

File: profiles/config.py [hunk 1]
```diff
@@ -9,29 +9,32 @@ class Config(ABC):
...
 
 
 class ContinuumTestConfig(ProdConfig):
-    pass
+    name = 'continuumtest'
 
 
 class End2EndConfig(ProdConfig):
-    pass
+    name = 'end2end'
```
Comment (thewilkybarkid) [inline profiles/config.py]: Could these match up with `ENVIRONMENTS` somehow?

File: profiles/api/errors.py [hunk 2]
```diff
@@ -25,22 +31,30 @@ def client_error_handler(exception: ClientError) -> Response:
...
+
     body = remove_none_values({
         'error': exception.error,
         'error_description': exception.description,
     })
 
     return make_response(jsonify(body), exception.status_code)
+
+
+def _handle_error(exception: HTTPException) -> Response:
```
Comment (thewilkybarkid) [inline profiles/api/errors.py]: `_handle_http_error`?

File: profiles/api/ping.py [hunk 3]
```diff
@@ -13,4 +13,10 @@ def _ping() -> Response:
 
         return response
 
+    @blueprint.route('/error')
+    def _error() -> Response:
```
Comment (thewilkybarkid) [inline profiles/api/ping.py]: Was this meant to be removed later?

File: profiles/config.py (hunk 1 above)
Comment (giorgiosironi) [reply profiles/config.py]: magic will follow

File: profiles/api/ping.py (hunk 3 above)
Comment (giorgiosironi) [reply profiles/api/ping.py]: useful for testing purposes, could be enabled only on some environments? Although, it's useful even to test the production deployment of logging

File: profiles/logging.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+import os
+import sys
+from typing import List
+
+from pythonjsonlogger import jsonlogger
+
+
+class CustomJsonFormatter(jsonlogger.JsonFormatter):
+    def formatException(self, ei: List) -> dict:
+        info_type, info_value, info_traceback = sys.exc_info()
```
Comment (thewilkybarkid) [inline profiles/logging.py]: Is this meant to not use `ei`?

Comment (giorgiosironi) [reply profiles/logging.py]: no, it should be unpack that rather than call again `sys.exc_info()`

**Pull request merged**